### ISSUE TITLE
Discussion: future of this project

### ISSUE THREAD

**Issue by Gekkio** [Gekkio/imgui-rs#292]
...discussion based on comments in #285 

Can everybody please clarify what exactly is expected of me that is not being provided at the moment? I'll happily consider giving full access to more people, but I'm also getting very mixed signals here.

@rasky wants review of this branch, which I'm not capable of delivering because this pull request has very little information about the change, and reviewing the pull request requires a good understanding of both the Dear Imgui feature (which is *still* experimental), and how that maps to imgui-rs.

What about @Jasper-Bekkers, @jaynus? What needs do you have that are not being met at the moment but would be if I gave full admin access to some people (you?)? My criteria for a crate being "unmaintained" is usually that pull requests don't get merged, which doesn't seem to be the case here, because the problem seems to be lack of review and/or lack of active development. I'm not actively doing work on this crate, but it's not the same thing as not accepting any changes.

There's nothing preventing somebody else from reviewing this pull request right here. In fact, if docking was in dear imgui master and somebody from Embark had reviewed this pull request, I'd merge it without hesitation.

There's nothing preventing somebody else from implementing "critical changes that weren't finished". If reasonable pull requests were submitted, I'd merge them.

And just to be clear: in any case, forking won't be needed. I'd much rather completely give away this project to somebody else than let it rot. However, the discussion so far seems to be mostly about credentials, not about the actual effort that moves the project forward. I've already given contributor access to one other person, and will gladly do it again. But what exactly does it change in practice other than credentials?

**TL;DR:**

"I need X but Gekkio doesn't do/provide Y" <- please fill in X and Y

"Q is not possible right now, but if Gekkio gave admin access to &lt;name of person&gt;, it would be possible" <- please fill in Q

**Comment by Jasper-Bekkers** [comment]
Hey @Gekkio, I think you're doing a pretty amazing job developing and maintaining imgui-rs, and we're happily using it (almost as-is). You mentioned you didn't have a lot of time on your hands to look at bigger PR's which is what I responded to, and wanted to help out with.

> In fact, if docking was in dear imgui master and somebody from Embark had reviewed this pull request, I'd merge it without hesitation.

This is along the lines I'm thinking tbh, the project itself looks like it won't need a lot of supervision (especially since Dear ImGui doesn't break back compat all that much), but if you're lacking time at the moment I'm happy to jump in and do some of that work with / for you.

**Comment by Gekkio** [comment]
Cool! :smile: 
I do appreciate help and am really excited about all the Rust work Embark is doing, so I've added you as a collaborator. Let me know if some other people from Embark would benefit from collaborator status as well and I'll add them.

> we're happily using it (almost as-is)

Out of curiosity, what kind of changes do you have on top of upstream imgui-rs? Any pain points that could be alleviated by upstream changes?

**Comment by jaynus** [comment]
@Gekkio I'd like to echo @Jasper-Bekkers comments, I was really responding to the exact same mention of not having time for reviewing larger PR's, as I'd like to also help keep this updated if needed. It is used by multiple projects in the amethyst org, as well as my amethyst-imgui crate, which is used by amethyst users. 

Since you have @Jasper-Bekkers as a collab now, and are at least willing to accept reviews from them, I'm happy with the resolution as well! I just wanted to make sure to be able to keep this project alive and moving forward.

**Comment by RoyJacobs** [comment]
Just curious, is there any update on this? I'd like very much to experiment with the `docking` branch (although there's definitely no rush), so I was just wondering.

**Comment by Mempler** [comment]
> Just curious, is there any update on this? I'd like very much to experiment with the docking branch (although there's definitely no rush), so I was just wondering.

same here but i don't want to experiment with it but rather use it in my engine.

**Comment by Gekkio** [comment]
Nothing has really changed:
* docking branch is still beta in upstream and therefore won't be supported here. Anybody with enough time and skill can maintain it in a fork without slowing down changes in the main imgui-rs repository
* overall I'm still maintaining imgui-rs quite randomly and not actively working on it. But AFAIK at the moment the only thing where my involvement is really needed is official crates.io releases

**Comment by luke-titley** [comment]
Hey guys. I'm pretty desperate for the docking branch and have the time (one week full-time), to pull it down and bring it across. Has anyone else made progress on this before I start looking into it?

**Comment by RoyJacobs** [comment]
I don't believe so, no. Great that you're planning to work on it!

**Comment by luke-titley** [comment]
okay great. I'll post progress updates, to the docking issue #249

**Comment by luke-titley** [comment]
Hey, I just want to comment on this further. The docking branch of dear imgui is super useful for people that want to use imgui as the main ui for their application. It does all the work of layout out your window correctly. I've showed a snap shot of a rust app, using it fine. I'm based off the master from this repo, with docking merged in.


![output](https://user-images.githubusercontent.com/55767239/93021384-9043d500-f5e2-11ea-8b09-73441a3abbf9.gif)

The thing is, to get this working, changes need to be made in imgui-rs. On top of that, there are multiple people interested in moving this forward, sorting out the safe api so that when this features goes to stable upstream, imgui-rs will be ready. It's a long lived feature that is in beta. It's not a short feature branch. In order to allow multiple people to collaborate on getting docking working in imgui-rs, with a rust like api, there needs to be a central place to push/pull. The options are:
* make a fork
* to add a branch to this repo.

In my view, currently, this is the canonical place to go to for rust bindings for imgui, it seems unnecessary for someone else to make a fork for this branch. It would fragment development. Please please please, say yes to adding a docking branch, and if you are super busy at the moment, please find someone who can be in charge of managing pull requests for this project.

**Comment by Boscop** [comment]
Is there a discord / telegram / gitter channel for imgui-rs? If not, it would be really great to have one! :)

**Comment by Boscop** [comment]
I just found out there is already a discord for the C++ version of imgui, with over 1000 members:
https://discord.gg/NgJ4SEP
Maybe @Gekkio can ask ocornut if we can have an `#imgui-rs` channel on that server?
(If it's too much hassle to setup a new discord server.)

**Comment by Gekkio** [comment]
`imgui-rs` is now looking for maintainer(s) :slightly_smiling_face:
I'll close this issue, but discussion can continue here:

https://github.com/Gekkio/imgui-rs/issues/392
